examples/interpreter-demo/interpreter.js

```diff
@@ -3831,17 +3826,14 @@ Interpreter.prototype['stepCallExpression'] = function(stack, state, node) {
 };
 
 Interpreter.prototype['stepCatchClause'] = function(stack, state, node) {
-  if (!state.done_) {
-    state.done_ = true;
-    // Create an empty scope.
-    var scope = this.createSpecialScope(state.scope);
-    // Add the argument.
-    this.setProperty(scope.object, node['param']['name'], state.throwValue);
-    // Execute catch clause.
-    return new Interpreter.State(node['body'], scope);
-  } else {
-    stack.pop();
-  }
+  // No need to hit this node again on the way back up the stack.
+  stack.pop();
+  // Create an empty scope.
+  var scope = this.createSpecialScope(state.scope);
+  // Add the argument.
+  this.setProperty(scope.object, node['param']['name'], state.throwValue);
+  // Execute catch clause.
+  return new Interpreter.State(node['body'], scope);
```

Comment:
As discussed: this can be inlined in `stepTryStatement`.

Reply:
Done.